src/java.desktop/unix/native/libpipewire/include/spa/utils/defs.h

```diff
@@ -58,7 +58,9 @@ extern "C" {
 #if defined(__clang__) && defined(__cplusplus) && __cplusplus >= 201103L
    /* clang's fallthrough annotations are only available starting in C++11. */
 #  define SPA_FALLTHROUGH [[clang::fallthrough]];
-#elif __GNUC__ >= 7 || __clang_major__ >= 10
+#elif defined(__GNUC__) && __GNUC__ >= 7
+#  define SPA_FALLTHROUGH __attribute__ ((fallthrough));
```

Comment:
Upstream still has this issue/bug. 
https://gitlab.freedesktop.org/pipewire/pipewire/-/blob/master/spa/include/spa/utils/defs.h
Do they intend to support only GNUC 7 and up ? Not clear. Best to ask.

Reply:
Ideally, this should be fixed upstream, but while we wait, we can try to fix the build failure in our source by adding definition of `__clang_major__` just before including the [pipewire/stream.h](https://github.com/openjdk/jdk/blob/master/src/java.desktop/unix/native/libawt_xawt/awt/screencast_pipewire.h#L36)

```
// some comment why we are doing this
#ifndef __clang_major__
#define __clang_major__ 0
#endif
```

It looks like that the `defs.h` is the only file that uses `__clang_major__`  in libxawt, so this define should be safe.


But the question of GCC < 7 support remains open.

Reply:
Um. If we want to minimize divergence with upstream, defining `__clang_major__` seems riskier than rewriting a single block that uses it incorrectly. Because I cannot be sure if _any other_ `ifdef` that checks `__clang_major__` would work correctly.